File: reto3.java
import java.util.Scanner;
import java.time.*;
import java.time.format.DateTimeFormatter;

public class reto3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        /*
         * Reto 3 abril - El tiempo
         * Enunciado
         * Crear dos variables utilizando los objetos fecha.
         * 
         * En la primera se representa la fecha (día, mes, año) actual.
         * En la segunda se representa la fecha de nacimiento.
         * Calcular cuántos años han transcurrido entre ambas fechas y muestra su
         * resultado de maneras diferentes Día, mes y año. Hora, minuto y segundo. Día
         * de año. Día de la semana. Nombre del mes.
         * 
         * Pistas: buscar sobre el paquete java.time para para fechas y horas.
         */

        System.out.println("Introduce un año formato YYYY: ");
        int year = scanner.nextInt();
        System.out.println("Introduce un mes MM: ");
        int month = scanner.nextInt();
        System.out.println("Introduce un dia DD: ");
        int day = scanner.nextInt();

        LocalDate fechaDada = LocalDate.of(year, month, day);
        LocalDate fechaActual = LocalDate.now();
        Period tiempoTranscurrido = Period.between(fechaDada, fechaActual);
        System.out.println("Han pasado: " + tiempoTranscurrido.getYears() + " años, "
                +
                tiempoTranscurrido.getMonths() + " meses y " + tiempoTranscurrido.getDays() +
                " días.");
        System.out
                .println(
                        fechaActual.getDayOfMonth() + " de " + fechaActual.getMonth() + " de " +
                                fechaActual.getYear());
        LocalDate iniciAno = LocalDate.of(2025, 1, 1);
        Period diasAnoTranscurrido = Period.between(fechaActual, iniciAno);
        System.out.println("Han transcurrido " + diasAnoTranscurrido.getDays() +
                " días desde el inicio de año");
        System.out.println("Hoy es " + fechaActual.getDayOfWeek());
        System.out.println("Estamos en el mes de  " + fechaActual.getMonth());
    }

}

File: reto3.py
'''Reto 3 abril - El tiempo
Enunciado
Crear dos variables utilizando los objetos fecha.

En la primera se representa la fecha (día, mes, año) actual.
En la segunda se representa la fecha de nacimiento.
Calcular cuántos años han transcurrido entre ambas fechas y muestra su resultado de maneras diferentes Día, mes y año. Hora, minuto y segundo. Día de año. Día de la semana. Nombre del mes.

Pistas: buscar sobre el paquete java.time para para fechas y horas.'''

from datetime import datetime, date, time

fecha_hoy = date.today()
fecha = input("Introduce una fecha en formato AAAA-MM-DD:")
fecha = date(int(fecha.split('-')[0]), int(fecha.split('-')[1]), int(fecha.split('-')[2]))
tiempo_transcurrido = fecha_hoy - fecha

year = tiempo_transcurrido.days/365
if (tiempo_transcurrido.days%365) >= 30:
    month = (tiempo_transcurrido.days%365)/30
    days = (tiempo_transcurrido.days%365)%30
elif (tiempo_transcurrido.days%365 <= 30) & (tiempo_transcurrido.days%365 > 0):
    days = tiempo_transcurrido.days%365
else:
    days = 0

print(f'Han pasado {round(year)} años, {round(month)} meses y {days} días')
print(format(fecha, '%d de %B de %Y'))
print(f'Son las {datetime.now().hour}:{datetime.now().minute}')
print(f'Han pasado {fecha_hoy-date(2025,1,1)}')
print(f'Hoy es {fecha_hoy.strftime('%A')}')
print(f'El mes es {fecha_hoy.strftime('%B')}') 